### DISCUSSION TITLE
GUI not working (mostly) on headless VM (+automation)

### DISCUSSION THREAD

**Discussion by dhstsw** [apohl79/audiogridder#451] [Help]
Hi,
first of all, thanks for the great work.

As for the title, i'm running a VM with Win10 64 in Unraid (so, KVM) - AudioGridder 1.1.1.
If i passthrough the GPU to the VM then AudioGridder (server) runs ok.
If i don't (so, when i need to access the VM i use remote desktop or VNC - or have AudioGridder Server 'autorun' on startup without any intervention) the the clients can connect and have the server load the plugin(s) but i'm mostly unable to open the editor (gui) of the plugin(s). It usually causes the server to crash/restart. Even when i manage to open the GUI it's not responsive and after a while the server disconnects/restarts.

Also automation seems not to be working anymore (Ableton Live 11 - I assign automation, in Live "configure", move the assignet automation parameter, nothing happens).

Thanks.

C.

**Comment by CaBuffalo** [comment]
Hi, what’s your DAW?

Does this happen to every plugin you open?

**Reply by dhstsw** [reply]
Yes.
Well, while i can live with passing thru the GPU, the main problem is the automation.
DAW is Live 11.

I know how automation works with it (and i can make it work with anything - closest plugin is Vienna, wich works).
So, in Audiogridder i assign the parameter i want to automate to an automation channnel, in Live i hit the "Configure" thing, i move the parameter (in the normal gui and/or in the generic editor) and live doesn't receive anything.

Any ideas?
Thanks.

**Comment by apohl79** [comment]
Try connecting via vnc first.

**Reply by dhstsw** [reply]
I'd love to try that but for some reason i can't connect to it with VNC anymore (and i guess that's the heart of problem #1).
:(

I have to investigate that further.

EDIT: made VNC work. The problem now is that when i launch Audiogridder it pops up an error windows stating what you see below.
![Immagine 2021-03-27 235654](https://user-images.githubusercontent.com/59790194/112736006-21afde80-8f58-11eb-9037-8f59763ee461.jpg)

**Reply by sparksandshadows** [reply]
I have the same issue

**Comment by jaacMusic** [comment]
I have dealt with this issue. You will need to either put a dummy display connector to your video card or install a virtual display adapter that allows you to run your PC without a physical video card but resolves the resolution issue. Sorry, I don't remember which virtual display driver I use but I did find a free one online using Google for search.

**Reply by sparksandshadows** [reply]
Are there any plans to patch this permanently so we don't need a workaround? There should at least be an official list somewhere of fixes for this common issue.

**Reply by jaacMusic** [reply]
Yeah, maybe Andreas could include something like an open-source driver option install for those that use the AG server without a monitor/video card. 
For now, I don't see it as an AG server issue; it just the way the AG server uses VNC technology to deliver the GUI. This VNC technology is directly attached to the video resolution of your system, which in this case you don't have because you don't have a monitor attached.
@apohl79 Hi Andreas, have you consider using a different approach for capturing  VST's GUI? Also, another thing. I have seen that when I tried to open two AG plugin GUI on a project it tends to crash the AG server because of the limitation of trying to open one GUI at a time.

**Reply by apohl79** [reply]
Well, there is multiple things to this. To reliably record the plugin UI ag uses screen capturing. But this requires a desktop session. This can be opened by logging in locally or remotely. For whatever reason RDP does not work properly on Windows, but VNC does. VNC itself does not have to do anything with ag.

Now, as the above is working, even if it is not perfect, improvements to this are rather low on the prio list.

Regarding multiple windows support. It would require some window management functionality to efficiently use and capture the available desktop space. Again, improvements are absolutely possible, but are relatively large effort, thus pretty low on my prio list.

**Reply by sparksandshadows** [reply]
I understand.. perhaps as the lowest effort solution the forum should simply have a pinned thread of software you can use (with instructions) to emulate the monitor for UI Screen Capturing on headless windows machines.

**Comment by DAMskill** [comment]
Ooh grate i want to start testing it on a proxmox server with 400gb of ram.

What about extra latency when running from vm?

Try a hdmi dummy. On ali you buy 5 for 5euro.
